File: Problemset/1276-Number of Burgers with No Waste of Ingredients/Mathematical.py
class Solution:
    def numOfBurgers(self, tomatoSlices: int, cheeseSlices: int) -> List[int]:
        if tomatoSlices % 2 == 0 \
                and tomatoSlices >= 2 * cheeseSlices \
                and tomatoSlices <= 4 * cheeseSlices:
            return [
                tomatoSlices // 2 - cheeseSlices,
                2 * cheeseSlices - tomatoSlices // 2
            ]
        return []

File: Problemset/1276-Number of Burgers with No Waste of Ingredients/Mathematical.rs
impl Solution {
    pub fn num_of_burgers(tomato_slices: i32, cheese_slices: i32) -> Vec<i32> {
        if tomato_slices % 2 == 0
            && tomato_slices >= 2 * cheese_slices
            && tomato_slices <= 4 * cheese_slices
        {
            vec![
                tomato_slices / 2 - cheese_slices,
                2 * cheese_slices - tomato_slices / 2,
            ]
        } else {
            vec![]
        }
    }
}
